redisson/src/main/java/org/redisson/extension/aws/ElasticCacheReplicationGroupServersConfig.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2016 Nikita Koksharov
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.redisson.extension.aws;
+
+import java.net.URL;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.redisson.config.BaseMasterSlaveServersConfig;
+
+import com.amazonaws.services.elasticache.AmazonElastiCache;
+
+
+public class ElasticCacheReplicationGroupServersConfig extends BaseMasterSlaveServersConfig<ElasticCacheReplicationGroupServersConfig> {
+
+	private String replicationGroupId;
+
+	private AmazonElastiCache awsElastiCacheClient;
+
+	private int scanInterval = 1000;
+	
+	/**
+     * Redis slave servers addresses
+     */
+    private Set<URL> slaveAddresses = new HashSet<URL>();
+
+    /**
+     * Redis master server address
+     */
+    private URL masterAddress;
+
+    /**
+     * Database index used for Redis connection
+     */
+    private int database = 0;
+
+	public ElasticCacheReplicationGroupServersConfig() {
+	}
+
+	public ElasticCacheReplicationGroupServersConfig setReplicationGroupId(String replicationGroupId) {
+		this.replicationGroupId = replicationGroupId;
+		return this;
+	}
+
+	public ElasticCacheReplicationGroupServersConfig setAmazonElastiCacheClient(AmazonElastiCache awsElastiCacheClient) {
+		this.awsElastiCacheClient = awsElastiCacheClient;
```

Comment:
For using internal client, we need to support various ways for providing aws credentials and different users have different preferences on how to supply aws key and secret.
http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default.
And some organisations use their preferred KMS systems apart from above defaults.  i thought it would be simpler if they can supply the client as per their preferred method.